### PR TITLE
Enabled writing geom. and data in different fp. formats

### PR DESCRIPTION
**Pull request by ravil-mobile**

### CODE REVIEW

File: XdmfWriter.h [hunk 1]
```diff
@@ -64,7 +65,7 @@ namespace xdmfwriter
 /**
  * Writes data in XDMF format
  */
-template<TopoType Topo, typename T>
+template<TopoType Topo, typename T, typename D>
```
Comment (sebwolf-de) [inline XdmfWriter.h:68]: Maybe you can also call these `VertexDataType` and `CellDataType` to make it more verbose.

**Review by sebwolf-de — COMMENTED** [review]
Hi Ravil,
no major issues, just some small questions about additional template parameters.
I'll have a look at the other PR in a second.

File: XdmfWriter.h (hunk 1 above)
Comment (uphoffc) [reply XdmfWriter.h:68]: You could also write typename D = T in order to not break existing code.

Comment (ravil-mobile) [reply XdmfWriter.h:68]: Thanks! I like both suggestions!

**Review by sebwolf-de — APPROVED** [review]

**Pull request merged**